transport/http_util.go

```diff
@@ -149,17 +155,43 @@ func isWhitelistedPseudoHeader(hdr string) bool {
 	}
 }
 
-func validContentType(t string) bool {
-	e := "application/grpc"
-	if !strings.HasPrefix(t, e) {
-		return false
+// getContentSubtype returns the content-subtype for the given content-type.
+// The given content-type must be a valid content-type that starts with
+// "application/grpc". A content-subtype will follow "application/grpc"
+// after a "+" or ";". See https://grpc.io/docs/guides/wire.html#requests
+// for more details.
+//
+// If contentType is not a valid content-type for gRPC, the boolean
+// will be false, otherwise true. If content-type == "application/grpc",
+// "application/grpc+", or "application/grpc;", the boolean will be true,
+// but no content-subtype will be returned.
+//
+// contentType is assumed to be lowercase already.
+func getContentSubtype(contentType string) (string, bool) {
```

Comment:
Go style nit: s/getC/c/

Reply:
Ugh do I really have to do this? It makes other things confusing IMO :-) I can if you want, no worries

Reply:
OK.  I think the hatred is more toward accessors and this is actively doing something.